test/prefer-key-over-key-code.js

```diff
@@ -0,0 +1,519 @@
+					if (a.key === 'Escape') {
+					}
+				});
+			`,
+			errors: [error('keyCode')]
+		},
+		{
+			code: `
+				foo.addEventListener('click', function(a, b, c) {
+					if (a.keyCode === 27) {
+					}
+				});
+			`,
+			output: `
+				foo.addEventListener('click', function(a, b, c) {
+					if (a.key === 'Escape') {
+					}
+				});
+			`,
+			errors: [error('keyCode')]
+		},
+		{
+			code: `
+				foo.addEventListener('click', function(b) {
+					if (b.keyCode === 27) {
+					}
+				});
+			`,
+			output: `
+				foo.addEventListener('click', function(b) {
+					if (b.key === 'Escape') {
+					}
+				});
+			`,
+			errors: [error('keyCode')]
+		},
+		{
+			code: `
+				foo.addEventListener('click', e => {
+					const {keyCode, a, b} = e;
+				});
+			`,
+			errors: [error('keyCode')]
+		},
+		{
+			code: `
+				foo.addEventListener('click', e => {
+					const {a: keyCode, a, b} = e;
+				});
+			`,
+			errors: [error('keyCode')]
+		},
+		{
+			code: `
+				add.addEventListener('keyup', event => {
+					f.addEventList('some', e => {
+						const {keyCode} = event;
+						console.log(event.key)
+					})
+				})
+			`,
+			errors: [error('keyCode')]
+		},
+		{
+			code: `
+				window.addEventListener('click', e => {
+					console.log(e.charCode);
+				})
+			`,
+			errors: [error('charCode')]
+		},
+		{
+			code: `
+				foo11111111.addEventListener('click', event => {
+					if (event.charCode === 27) {
+					}
+				});
+			`,
+			output: `
+				foo11111111.addEventListener('click', event => {
+					if (event.key === 'Escape') {
+					}
+				});
+			`,
+			errors: [error('charCode')]
+		},
+		{
+			code: `
+				foo.addEventListener('click', a => {
+					if (a.charCode === 27) {
+					}
+				});
+			`,
+			errors: [error('charCode')],
+			output: `
+				foo.addEventListener('click', a => {
+					if (a.key === 'Escape') {
+					}
+				});
+			`
+		},
+		{
+			code: `
+				foo.addEventListener('click', (a, b, c) => {
+					if (a.charCode === 27) {
+					}
+				});
+			`,
+			errors: [error('charCode')],
+			output: `
+				foo.addEventListener('click', (a, b, c) => {
+					if (a.key === 'Escape') {
+					}
+				});
+			`
+		},
+		{
+			code: `
+				foo.addEventListener('click', function(a, b, c) {
+					if (a.charCode === 27) {
+					}
+				});
+			`,
+			output: `
+				foo.addEventListener('click', function(a, b, c) {
+					if (a.key === 'Escape') {
+					}
+				});
+			`,
+			errors: [error('charCode')]
+		},
+		{
+			code: `
+				foo.addEventListener('click', function(b) {
+					if (b.charCode === 27) {
+					}
+				});
+			`,
+			errors: [error('charCode')],
+			output: `
+				foo.addEventListener('click', function(b) {
+					if (b.key === 'Escape') {
+					}
+				});
+			`
+		},
+		{
+			code: `
+				foo.addEventListener('click', e => {
+					const {charCode, a, b} = e;
+				});
+			`,
+			errors: [error('charCode')]
+		},
+		{
+			code: `
+				foo.addEventListener('click', e => {
+					const {a: charCode, a, b} = e;
+				});
```

Comment:
Can you also add a test with destructuring directly in the parameter?

```js
foo.addEventListener('click', ({keyCode}) => {

});
```

Reply:
Nice catch, I did not handle this case :D. Added now!